src/en/darkscience/src/eu/kanade/tachiyomi/extension/en/darkscience/DarkScience.kt

```diff
@@ -0,0 +1,155 @@
+    }
+
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> = Observable.just(
+        MangasPage(
+            listOf(initTheManga(SManga.create())),
+            false,
+        ),
+    )
+
+    // We still (re-)initialise all properties here, for this method also gets called on a
+    // backup restore. And in a backup, only `url` and `title` are preserved.
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> = Observable.just(initTheManga(manga))
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> =
+        accumulateChapters(Triple(client.newCall(chapterListRequest(manga)), 0.0F, Observable.empty())).concatMap { it.third.toList() }
+
+    override fun pageListParse(response: Response): List<Page> =
+        listOf(
+            Page(
+                0,
+                "",
+                response
+                    .asJsoup()
+                    .selectFirst("article.post img.aligncenter")!!
+                    .attr("src"),
+            ),
+        )
+
+    override fun imageUrlParse(page: Response): String =
+        page.asJsoup().selectFirst("article.post img.aligncenter")!!.attr("src")
+
+    override fun popularMangaRequest(page: Int): Request = throw UnsupportedOperationException()
+
+    override fun popularMangaParse(response: Response): MangasPage = throw UnsupportedOperationException()
+
+    override fun mangaDetailsParse(response: Response): SManga = throw UnsupportedOperationException()
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> = throw UnsupportedOperationException()
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request = throw UnsupportedOperationException()
+
+    override fun searchMangaParse(response: Response): MangasPage = throw UnsupportedOperationException()
+
+    override fun latestUpdatesRequest(page: Int): Request = throw UnsupportedOperationException()
+
+    override fun latestUpdatesParse(response: Response): MangasPage = throw UnsupportedOperationException()
+
+    override fun chapterListParse(response: Response): List<SChapter> = throw UnsupportedOperationException()
+
+    private fun parseChapter(ch: Element, chLast: Float): SChapter {
+        val chTitle = ch.text()
+        val chLink = ch.attr("href")
+        val chDateMatch = chapterDateRegex.find(chLink)!!
+        val chNumMatch = chapterNumberRegex.find(chTitle)
+        val chDate = DATE_FMT.parse(chDateMatch.groupValues[1])?.time ?: 0L
+        val chNum = chNumMatch?.groupValues?.getOrNull(1)?.toFloat() ?: (chLast - 0.01F)
+
+        return SChapter.create().apply {
+            name = chTitle
+            chapter_number = chNum
+            date_upload = chDate
+            setUrlWithoutDomain(chLink)
+        }
+    }
+
+    private fun accumulateChapters(state: Triple<Call?, Float, Observable<SChapter>>): Observable<Triple<Call?, Float, Observable<SChapter>>> {
```

Comment:
you don't need to use observables, since the fetch methods are already run in a background thread. 

\+ observables are deprecated and will be removed in the next version of source api, so I would like to avoid new usage of them

Reply:
Oh interesting! And glad to hear observables’ll be nuked. They’re really awkward to use. I’ll revert this, then.